docs/REST/authentication.md

```diff
@@ -289,3 +289,62 @@ static class HttpContentExtensions
     }
 }
 ```
+### Java
+```java
+public CloseableHttpResponse execute(HttpUriRequest request, String credential, String secret)
+        throws IOException, URISyntaxException {
+    Map<String, String> authHeaders = sign(request, credential, secret);
+    authHeaders.forEach(request::setHeader);
```

Comment:
>authHeaders.forEach(request::setHeader); [](start = 4, length = 40)

We had passed request into sign(), shouldn't we just set header inside it? Or call the method getAuthHeader()?

Reply:
This code is based off how the provider does it. It's mainly for readability where the `authHeaders` are generated, then attached to the request.

Reply:
sign() function name doesn't clearly describe the purpose because it didn't sign it yet but generate the header for you. People need to read the code to know this.

---
In reply to: [297318652](https://github.com/Azure/AppConfiguration/pull/93#discussion_r297318652) [](ancestors = 297318652)

Reply:
I updated it from sign() to generateHeader()